api/src/main/java/com/velocitypowered/api/event/player/PlayerClientBrandEvent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2018 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.event.player;
+
+import com.velocitypowered.api.proxy.Player;
+
+/**
+ * Fired when a {@link Player} sends the <code>minecraft:brand</code> plugin message.
+ */
+public class PlayerClientBrandEvent {
```

Comment:
In tandem with preconditions (see other comment), a final class prevents sub-classes from overriding methods and breaking contracts. Other events do this as well.
```suggestion
public final class PlayerClientBrandEvent {
```